### PR TITLE
Move IDE-related projects to the IDE platform directory

### PR DESCRIPTION
**Pull request by hegyibalint**
This PR moves some of the IDE-team owned project residing in the `subprojects` directory to the `projects/ide` directory.

### CODE REVIEW

**Comment by hegyibalint** [comment]
@bot-gradle test this

File: .cm/platform_labels.cm [hunk 1]
```diff
@@ -150,10 +150,13 @@ platforms:
...
     subprojects:
-      - 'platforms/ide/'
-      - 'subprojects/ide/'
-      - 'subprojects/tooling-api/'
-      - 'subprojects/tooling-api-builders/'
+      - 'platforms/ide/base-ide-plugins'
+      - 'platforms/ide/ide'
+      - 'platforms/ide/ide-native/'
+      - 'platforms/ide/ide-plugins/'
+      - 'platforms/ide/problems/'
```
Comment (tresat) [inline .cm/platform_labels.cm:157]: ❗ We have this marked as a `Core/runtime` project in the Code Areas sheet, which seems correct to me - this is not merely IDE-related code.  Did this get change?

**Review by tresat — CHANGES_REQUESTED** [review]

File: .cm/platform_labels.cm (hunk 1 above)
Comment (tresat) [reply .cm/platform_labels.cm:157]: Discussed, this is the temporary home during development.  Please add a comment to tjhe platform in `CODEOWNERS` to note this.

**Comment by hegyibalint** [comment]
> ❗ Please update `summary_table.cm` with the same platform changes as well.

Done, PTAL

File: .cm/platform_labels.cm [hunk 2]
```diff
@@ -150,10 +150,13 @@ platforms:
   - ide:
     name: 'ide'
     subprojects:
-      - 'platforms/ide/'
-      - 'subprojects/ide/'
-      - 'subprojects/tooling-api/'
-      - 'subprojects/tooling-api-builders/'
+      - 'platforms/ide/base-ide-plugins'
```
Comment (tresat) [inline .cm/platform_labels.cm:153]: ❗ The change to `summary_table.cm` is correct.  Make the change to this file match and it'll be good to go.

**Review by tresat — CHANGES_REQUESTED** [review]

File: .cm/platform_labels.cm (hunk 2 above)
Comment (hegyibalint) [reply .cm/platform_labels.cm]: Done

**Review by tresat — APPROVED** [review]

**Review by donat — APPROVED** [review]

**Comment by hegyibalint** [comment]
@bot-gradle test and merge please

**Pull request merged**